consumer.go

```diff
@@ -479,23 +479,27 @@ func (child *partitionConsumer) parseResponse(response *FetchResponse) ([]*Consu
 	prelude := true
 	var messages []*ConsumerMessage
 	for _, msgBlock := range block.MsgSet.Messages {
-
 		for _, msg := range msgBlock.Messages() {
-			if prelude && msg.Offset < child.offset {
+			offset := msg.Offset
+			if child.conf.Version.IsAtLeast(V0_10_0_0) {
```

Comment:
If I'm reading the docs properly, this condition should be based on `msg.Msg.Version` since 0.10 brokers can still use the old format in some configurations.